radixdlt-java/src/main/java/com/radixdlt/client/core/crypto/RadixECKeyPairs.java

```diff
@@ -0,0 +1,63 @@
+package com.radixdlt.client.core.crypto;
+
+import org.radix.crypto.Hash;
+
+import java.util.Arrays;
+
+public class RadixECKeyPairs {
+
+    static {
+        // Make sure the BouncyCastle cryptographic implementation is properly
+        // installed.
+        ECKeyPairGenerator.install();
+    }
+
+    public static RadixECKeyPairs newInstance() {
+        return new RadixECKeyPairs();
+    }
+
+
+    // Intentionally hidden constructor
+    private RadixECKeyPairs() {
+    }
+
+    /**
+     * Generates a new, deterministic {@code ECKeyPair} instance based on the
+     * provided seed.
+     *
+     * @param seed The seed to use when deriving the key pair instance.
+     * @return A key pair that corresponds to the provided seed.
+     * @throws IllegalArgumentException if the seed is empty or a null pointer.
+     */
+    public ECKeyPair generateKeyPairFromSeed(byte[] seed) {
+        if (seed == null) {
+            throw new IllegalArgumentException("Seed must not be null");
```

Comment:
We've been using `Objects.requireNonNull` for null checks. Can we replace that here?

Reply:
Sure! I was aiming for consistency with the next block (checking for empty seed), but I can't rectify my choice beyond that.